src/lib/grammar_ast.rs

```diff
@@ -0,0 +1,171 @@
+use std::collections::{HashMap, HashSet};
+use std::fmt;
+
+pub struct GrammarAST {
```

Comment:
Ah okay. That makes kind of sense then. I guess my association with AST is too biased.